File: c/array/maximum-subarray.c
int maxSubArray(int* nums, int numsSize) {
    int i;
    int ret = nums[0];
    int cur = 0;

    for(i = 0; i < numsSize; ++i){
        cur += nums[i];
        ret = cur > ret ? cur : ret;
        cur = cur > 0 ? cur : 0;
    }

    return ret;
}

File: go/array/maximum-subarray.go
func maxSubArray(nums []int) int {
    var maxsum, cursum int

    maxsum = math.MinInt32
    for _, value := range nums {
        cursum += value
        if cursum > maxsum {
            maxsum = cursum
        }
        if cursum < 0 {
            cursum = 0
        }
    }

    return maxsum
}
